Rename SpanContext to SpanReference. (#1075)
open-telemetry/opentelemetry-specification: specification/metrics/api.md

Instruments are classified in several ways that distinguish them from
one another.

> 1. Synchronicity: A synchronous instrument is called by the user in a distributed [Context](../context/context.md) (i.e., SpanReference, Baggage). An asynchronous instrument is called by the SDK once per collection interval, lacking a Context.

Reviewer: "called in a Baggage" or "called in a SpanReference" sounds really strange. What is more, since the introduction of propagated span, it's actually a full span.

```suggestion
1. Synchronicity: A synchronous instrument is called by the user in a distributed [Context](../context/context.md) (i.e., with associated Span, Baggage, etc.). An asynchronous instrument is called by the SDK once per collection interval, lacking a Context.
```

Author: I feel this is scope creep - I'm merely updating what we had before. A follow up PR to clean up references to SpanContext -> Span (when using Propagated Span, that is) is due, but I'd even feel weird mixing concerns here.

| ValueObserver     | No  | No  | No  |

The synchronous instruments are useful for measurements that are

> gathered in a distributed [Context](../context/context.md) (i.e., SpanReference, Baggage).  The asynchronous instruments are

Reviewer: ```suggestion
gathered in a distributed [Context](../context/context.md) (i.e., with associated Span, Baggage, etc.).  The asynchronous instruments are
```

- [resources](../resource/sdk.md) associated with the SDK at startup.

Synchronous events have one additional property, the distributed

> [Context](../context/context.md) (i.e., SpanReference, Baggage)

Reviewer: ```suggestion
[Context](../context/context.md) (containing Span, Baggage, etc.)
```

CC @jmacd This has been that way for a long time and I'm not sure that really is OK. When I was originally proposing to store the full parent Context in the span in #875, I was shot down due to memory concerns.

Not related to this PR but seems like it must be fixed.

### Synchronous and asynchronous instruments compared

Synchronous instruments are called inside a request, meaning they

> have an associated distributed [Context](../context/context.md) (i.e., SpanReference, Baggage).  Multiple metric events may occur for a

Reviewer: ```suggestion
have an associated distributed [Context](../context/context.md) (with, Span, Baggage, etc.).  Multiple metric events may occur for a
```

Reviewer 2: ```suggestion
have an associated distributed [Context](../context/context.md) (with Span, Baggage, etc.).  Multiple metric events may occur for a
```

Synchronous measurements are implicitly associated with the
distributed [Context](../context/context.md) at runtime, which may

> include a SpanReference and Baggage entries.  The Metric SDK may use

Reviewer: ```suggestion
include a Span and Baggage entries. The Metric SDK may use
```
